BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c

```diff
@@ -100,8 +103,17 @@ PciDevicePresent (
   if (PciExpressRead16 (Address) == 0xFFFF) {
     return EFI_NOT_FOUND;
   } else {
-    CopyMem (Pci, (VOID *) ((UINTN)PcdGet64 (PcdPciExpressBaseAddress) + \
-                            PCI_EXPRESS_LIB_ADDRESS (Bus, Device, Func, 0)), sizeof (PCI_TYPE00));
+    //
+    // We perform a memory copy here manually to ensure DWORD alignment
+    // as required for PCI config space MMIO accessing. Using the
+    // BaseMemoryLibSse2 CopyMem routine can cause issue since it may
+    // fall back to byte-level copying once the DQWORD copying finishes.
```

Comment:
I think the issue is more on access unit greater than DWORD.   BYTE access for PCI space should be OK.
Since SSE2 might use 128bit access,  this might be the issue.

Reply:
Updated comment, maybe it is a more accurate description now?